spark-testing/spark_utils.py

```diff
@@ -127,10 +127,10 @@ def submit_job(
 
     conf_args = args.copy()
 
-    conf_args += [
-        '--conf spark.driver.memory=2g',
-        '--conf spark.mesos.role={}'.format(driver_role)
-    ]
+    conf_args += ['--conf spark.mesos.role={}'.format(driver_role)]
+
+    if "spark.driver.memory" not in conf_args:
+        conf_args += ['--conf spark.driver.memory=2g']
```

Comment:
Considering the check that we are performing above, let's be consistent with how we are adding these args. i.e. let's use:
```python
conf_args += ['--conf', 'spark.driver.memory=2g', ]
```
instead.